msvc/Custom-vs.cmake

```diff
@@ -1,28 +1,93 @@
 # CUSTOM PROPERTIES TO SET
 # GLOBAL
-
 set(CMAKE_VERBOSE_MAKEFILE ON)
-set(BUILD_STATIC_LIBRARY ON)
-set(USE_OPEN_MP OFF)
-set(TCL_VERSION 8.5)
-
-# Disable the following warnings in msvc
-# C4244 loss of data in conversion
-# C4267 loss of data in conversion
-# C4005 macro refinitions
-# C4996 unsafe functions
-# C4047 levels of indirection difference (int and void*)
-# C4090 different const qualifiers
-# C4477 format string type differences
-set(CMAKE_C_FLAGS "${CMAKE_CX_FLAGS} /wd4244 /wd4267 /wd4005 /wd4996 /wd4047 /wd4090 /wd4477")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4244 /wd4267 /wd4005 /wd4996 /wd4047 /wd4090 /wd4477")
 
+# Csound build options for MSVC
+set(BUILD_BUCHLA_OPCODES ON)
+set(BUILD_CHUA_OPCODES ON)
+set(BUILD_CSBEATS OFF) # Disabled due to flex issue
+set(BUILD_CSOUNDVST OFF)
+set(BUILD_CSOUND_AC OFF)
+set(BUILD_CSOUND_AC_LUA_INTERFACE OFF)
+set(BUILD_CSOUND_AC_PYTHON_INTERFACE OFF)
+set(BUILD_CUDA_OPCODES OFF)
+set(BUILD_CXX_INTERFACE ON)
+set(BUILD_DSSI_OPCODES OFF)
+set(BUILD_EMUGENS_OPCODES ON)
+set(BUILD_EXCITER_OPCODES ON)
+set(BUILD_FAUST_OPCODES OFF)
+set(BUILD_FLUID_OPCODES OFF)
+set(BUILD_FRAMEBUFFER_OPCODES ON)
+set(BUILD_HDF5_OPCODES ON)
+set(BUILD_IMAGE_OPCODES ON)
+set(BUILD_INSTALLER ON)
+set(BUILD_JACK_OPCODES OFF)
+set(BUILD_JAVA_INTERFACE ON)
+set(BUILD_LINEAR_ALGEBRA_OPCODES ON)
+set(BUILD_LUA_INTERFACE ON)
+set(BUILD_LUA_OPCODES ON)
+set(BUILD_MULTI_CORE ON)
+set(BUILD_OPENCL_OPCODES OFF)
+set(BUILD_OSC_OPCODES ON)
+set(BUILD_P5GLOVE_OPCODES OFF)
+set(BUILD_PADSYNTH_OPCODES OFF)
 set(BUILD_PD_CLASS OFF)
+set(BUILD_PLATEREV_OPCODES ON)
+set(BUILD_PYTHON_INTERFACE ON)
+set(BUILD_PYTHON_OPCODES ON)
+set(BUILD_RELEASE ON)
+set(BUILD_SCANSYN_OPCODES ON)
+set(BUILD_SELECT_OPCODE ON)
+set(BUILD_SERIAL_OPCODES ON)
+set(BUILD_STACK_OPCODES ON)
+set(BUILD_STATIC_LIBRARY ON)
+set(BUILD_STK_OPCODES ON)
+set(BUILD_TESTS ON)
+set(BUILD_UTILITIES ON)
+set(BUILD_VIRTUAL_KEYBOARD ON)
+set(BUILD_WEBSOCKET_OPCODE ON)
+set(BUILD_WIIMOTE_OPCODES OFF)
+set(BUILD_WINDSOUND ON)
+set(NEED_PORTTIME OFF)
+
+# Csound use features
+set(USE_ALSA 0) # N/A
+set(USE_ATOMIC_BUILTIN 0) # Needs code changes for MSVC
+set(USE_AUDIOUNIT 0) # N/A
+set(USE_CURL 0) # Not working, necessary even?
```

Comment:
I saw this and other uses of USE_CURL that were introduced into code. I think if we're going to have an option to explicitly turn of use of CURL, we can do it by check USE_CURL in CMake and optionally not set -DHAVE_CURL rather than adding USE_CURL everywhere in source.  I'd rather see this removed or updated before any merge.